lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -181,7 +181,16 @@ public IndexSearcher(IndexReader r) {
   }
 
   /** Runs searches for each segment separately, using the
-   *  provided Executor. NOTE:
+   *  provided Executor. The passed in Executor will also be
+   *  used by LRUQueryCache (if enabled) to perform asynchronous
+   *  query caching.
+   *  If a task is rejected by the host Executor, the failed task
+   *  will then be executed on the caller thread. This is done to
+   *  ensure that a query succeeds, albeit with a higher latency.
+   *  If a user wishes to modify the said behaviour, they can either
+   *  handle the exception in the provided Executor, or override
+   *  the said method in a custom extension of IndexSearcher.
```

Comment:
it's unclear to me which method you are recommending overriding?

Reply:
Removed